### PR TITLE
Processorpath

### PR DESCRIPTION
**Pull request by mernst**
This pull request is intended to correct issue #700, "Too many classes on the bootclasspath".

### CODE REVIEW

File: framework/src/org/checkerframework/framework/util/CheckerMain.java [hunk 1]
```diff
@@ -287,6 +309,38 @@ protected static File extractFileArg(final String argumentName, final File alter
...
+    protected static List<String> extractPpOpts(final List<String> args) {
+        List<String> actualArgs = new ArrayList<String>();
+
+        String path = null;
+
+        int i = 0;
+        while (i < args.size()) {
+
+            if (args.get(i).equals("-processorpath")) {
+                if (args.size() > i) {
```
Comment (smillst) [inline framework/src/org/checkerframework/framework/util/CheckerMain.java]: Should be `args.size() > i + 1`, right?  Would also be easier to spot the difference from the loop predicate  if it was `i + 1 < arg.size()`

File: framework/src/org/checkerframework/framework/util/CheckerMain.java [hunk 2]
```diff
@@ -123,7 +135,17 @@ public void addToRuntimeBootclasspath(List<String> runtimeBootClasspathOpts) {
...
-        extractedOps.add(0, this.checkersJar.getAbsolutePath());
+        extractedOps.add(0, this.checkerQualJar.getAbsolutePath());
+        return extractedOps;
+    }
+
+    // Assumes that createCpOpts has already been run.
+    protected List<String> createPpOpts(final List<String> argsList) {
+        final List<String> extractedOps = extractPpOpts(argsList);
+        if (extractedOps.isEmpty()) {
+            extractedOps.addAll(this.cpOpts);
```
Comment (smillst) [inline framework/src/org/checkerframework/framework/util/CheckerMain.java]: I don't understand why the classpath is added to the processor path if no explicit processor path is found.  (Or maybe that's not what's happening here?)

Comment (mernst) [inline framework/src/org/checkerframework/framework/util/CheckerMain.java]: That's what is happening.  I added a comment explaining why.

File: framework/src/org/checkerframework/framework/util/CheckerMain.java (hunk 1 above)
Comment (mernst) [inline framework/src/org/checkerframework/framework/util/CheckerMain.java]: You are right.  I copied that from the extractCpOpts routine, which contained this bug.  I will fix both routines.  Thanks!

**Comment by mernst** [comment]
Replaced by #761

**Pull request closed without merging**